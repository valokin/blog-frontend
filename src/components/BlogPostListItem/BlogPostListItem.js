import styles from './BlogPostListItem.module.css';

const BlogPostListItem = ({ post, onClick }) => {
  return (
    <div onClick={() => {
      onClick()
    }}
      className={styles.container}
      data-testid="blog-post-list-item"
    >
      <div className={styles.title}
        data-testid="blog-post-list-item-title"
        >
        {post.title}
      </div>
      <div className={styles.description}
        data-testid="blog-post-list-item-description"
      >
        {post.description}
      </div>
      <div className={styles.commentCount}
        data-testid="blog-post-list-item-comment-count"
      >
        {`Number of comments: ${post.comments.length}`}
      </div>
    </div>
  )
};

export default BlogPostListItem;