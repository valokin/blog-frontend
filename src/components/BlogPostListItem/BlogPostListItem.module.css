.container {
    height: 40vh;
    min-height: 200px;
    min-width: 300px;
    background-color: skyblue;
    margin: 6px;
    padding: 5%;
    position: relative;
    cursor: pointer;
}

.container:hover {
    opacity: 0.8;
}

.title {
    font-size: 32px;
    padding-bottom: 10px;
}

.description {
    font-size: 20px;
}

.commentCount {
    position: absolute;
    bottom: 3%;
    right: 3%;
    font-size: 22px;
}

