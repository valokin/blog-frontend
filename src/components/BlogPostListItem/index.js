import BlogPost from './BlogPostListItem';

export default BlogPost