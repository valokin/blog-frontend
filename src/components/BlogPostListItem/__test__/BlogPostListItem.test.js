import React from 'react';
import ReactDom from 'react-dom';
import BlogPostListItem from '../BlogPostListItem';
import PostWithComments from '../../../mocks/postWithComments.json';

import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

afterEach(cleanup);

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDom.render(<BlogPostListItem post={
        PostWithComments
    }></BlogPostListItem>, div);
});

it('renders blog post lits item correctly', () => {
    const {getByTestId } = render(<BlogPostListItem post={
        PostWithComments
    }></BlogPostListItem>) 
    expect(getByTestId('blog-post-list-item-title'))
        .toHaveTextContent('title 1');
    expect(getByTestId('blog-post-list-item-description'))
        .toHaveTextContent('description 1');
    expect(getByTestId('blog-post-list-item-comment-count'))
        .toHaveTextContent('Number of comments: 3');
})

it('matches snapshot', () => {
    const tree = renderer.create(<BlogPostListItem post={
        PostWithComments
    }></BlogPostListItem>).toJSON();

    expect(tree).toMatchSnapshot();
})