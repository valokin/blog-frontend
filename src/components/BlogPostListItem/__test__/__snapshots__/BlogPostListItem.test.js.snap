// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="container"
  data-testid="blog-post-list-item"
  onClick={[Function]}
>
  <div
    className="title"
    data-testid="blog-post-list-item-title"
  >
    title 1
  </div>
  <div
    className="description"
    data-testid="blog-post-list-item-description"
  >
    description 1
  </div>
  <div
    className="commentCount"
    data-testid="blog-post-list-item-comment-count"
  >
    Number of comments: 3
  </div>
</div>
`;
